### ISSUE TITLE
Keeps saying stop the app manually and nothing happens ://

### ISSUE THREAD

**Issue by shailantani** [xHookman/IGexperiments#17]
I installed instagram apk from apkmirror. And then I tried clicking kill and launch instagram button on IGexperiments app, but it kept saying stop the app manually ( I've given the lsposed and magisk rights).
And when i stop the app, and launch nothing happens.
Would be great if someone could help me out :))

**Comment by ReSo7200** [comment]
Hey, can you please confirm if you installed one of the supported Instagram versions?

**Comment by shailantani** [comment]
Yep, I did. it was the 305 version

**Comment by ReSo7200** [comment]
oh I ran into the same issue now, however dev options were enabled, thx for reporting this!

can you please try to open Instagram, click and hold on the home button?

**Comment by ReSo7200** [comment]
Hey again, so I the issue appears to be with the new Instagram versions.

The tool would still be able to enable developer options, but can't force stop and start Instagram app automatically.

I will try to push an update with a fix today.

btw, can u plz confirm if the developer options were enabled? (click and hold on the Instagram home button to check that)

**Comment by ReSo7200** [comment]
So I found where the issue is, I will push a new pull request right away!

Here's what were causing the issue:

Instagram older versions =< 302:

![image](https://github.com/xHookman/IGexperiments/assets/47777771/315f340f-e16a-47cc-a1ec-0d995df95bc3)

New Instagram versions > 302:

![image](https://github.com/xHookman/IGexperiments/assets/47777771/217fc294-6490-40fd-8196-71ae018edef1)

**Comment by shailantani** [comment]
> Hey again, so I the issue appears to be with the new Instagram versions.
> 
> The tool would still be able to enable developer options, but can't force stop and start Instagram app automatically.
> 
> I will try to push an update with a fix today.
> 
> btw, can u plz confirm if the developer options were enabled? (click and hold on the Instagram home button to check that)

Heyyy, sorry for the delay, but the dev options werent enabled for me for some reason :((
Do you want me to try out something?

**Comment by ReSo7200** [comment]
Have you tried to force stop the Instagram app?

**Comment by shailantani** [comment]
> Have you tried to force stop the Instagram app?

yepp, I did. So i installed the apk, and signed in and then force stopped it. And then I opened the IGexperiment app, and then opened instagram, but nothing seemed to happen

**Comment by ReSo7200** [comment]
hmm, no you have to install the Instagram app first, open IGexperiments app then force close Instagram, sign in and you should be able to see dev options.

**Comment by ReSo7200** [comment]
oh and another thing, you should add the Instagram app to the module scope!

**Comment by shailantani** [comment]
> hmm, no you have to install the Instagram app first, open IGexperiments app then force close Instagram, sign in and you should be able to see dev options.

So the thing is, after installing app and then opening IGexperiments, I dont get the option to force close the app. The force close app option appears only after I sign in.
Heres a screenshot after I just installed the app.

![WhatsApp Oct 8 Image](https://github.com/xHookman/IGexperiments/assets/98448265/9cde35ac-5cb9-41a3-934b-3c883dbe1d29)

**Comment by shailantani** [comment]
> oh and another thing, you should add the Instagram app to the module scope!

Oh yeah its already added through lspatch:

![WhatsApp Oct 8 Image (1)](https://github.com/xHookman/IGexperiments/assets/98448265/a68e390a-84d9-44ba-94ea-8540e6cd21c1)

**Comment by ReSo7200** [comment]
It's not about signing in, it's about launching the app, but that shouldn't be a problem...

I will push a new update in under 5 minutes or so.

will send you a testing apk, test it and tell me what happens.

**Comment by ReSo7200** [comment]
here you go!

can you please test it?

https://github.com/ReSo7200/IGexperiments/releases/tag/hotfix

**Comment by shailantani** [comment]
> here you go!
> 
> can you please test it?
> 
> https://github.com/ReSo7200/IGexperiments/releases/tag/hotfix

sure gimme a minute 😄

**Comment by ReSo7200** [comment]
note that it would show the same toast of "Stop the app manually", but it would force close and run Instagram.

first time it will close it self, click on kill and launch Instagram button again.

this is for testing only I will fix the toast issue later.

**Comment by shailantani** [comment]
> here you go!
> 
> can you please test it?
> 
> https://github.com/ReSo7200/IGexperiments/releases/tag/hotfix
Hey so the kill and launchworks but the thing is dev options werent enabled.
I installed the IGexperiments, then installed the instagram app.  then I pressed the kill and launch instagram, which worked. But the dev options werent enabled. Any idea?

**Comment by ReSo7200** [comment]
Strange...

I just tested it on 2 rooted devices and 1 non rooted it works just fine.

Can you check the logs of lsposed? Please clear the logs of lsposed, try igexperiments and then check the logs.

**Comment by shailantani** [comment]
> Strange...
> 
> I just tested it on 2 rooted devices and 1 non rooted it works just fine.
> 
> Can you check the logs of lsposed? Please clear the logs of lsposed, try igexperiments and then check the logs.

I dont know, seems to be weird. I might be doing something wrong 😭 
Here are the logs:
[LSPosed Oct 8.zip](https://github.com/xHookman/IGexperiments/files/12840359/LSPosed.Oct.8.zip)
Hope it was helpful ://

**Comment by shailantani** [comment]
Hey so i just redownloaded both IGexperiments and instagram. Seemed to work now
Seems weird, I'll try to check what went wrong
Thanks for all the help btw, I'll let you know if I find a bug

**Comment by ReSo7200** [comment]
Thanks!

I will close this once I open a pull request.

**Comment by shailantani** [comment]
> Thanks!
> 
> I will close this once I open a pull request.

Sure, by the way. About updating the app, I read the readme, but I still had a doubt.
After installing the new apk and patching it through lsposed. It asks to uninstall and install the new apk. So is that supposed to happen? Or is it supposed to just update the old to new apk without deleting the app?

**Comment by ReSo7200** [comment]
Hmm strange, I guess for rooted users you don't have to patch anything.

After enabling developer options, you have to enable the options noted in readme.

Disable IGexpeirments from lsposed (so it doesn't try to hook into a non existed method in the new instagram)

Download and install a new version.

And that's it.

**Comment by shailantani** [comment]
> Hmm strange, I guess for rooted users you don't have to patch anything.
> 
> After enabling developer options, you have to enable the options noted in readme.
> 
> Disable IGexpeirments from lsposed (so it doesn't try to hook into a non existed method in the new instagram)
> 
> Download and install a new version.
> 
> And that's it.

Perfect! Thanks a lot!
